### PR TITLE
Implement Trust Authentication

### PR DESCRIPTION
**Pull request by AndrewJackson2020**
This PR implements trust (no password authentication). This is somewhat related to #748 and has some overlapping changes but this PR does not require any changes to the dockerfile/CI.

This feature reflects postgres' implementation documented [here](https://www.postgresql.org/docs/current/auth-trust.html).

### CODE REVIEW

**Comment by AndrewJackson2020** [comment]
@drdrsh Any chance you can review this PR? It's tested, doesn't require any changes to the CI docker image, and has no breaking changes.

File: src/messages.rs [hunk 1]
```diff
@@ -64,6 +64,20 @@ where
     write_all(stream, auth_ok).await
 }
 
+/// Tell the client to use clearr text auth
+pub async fn clear_text_challenge<S>(stream: &mut S) -> Result<(), Error>
```
Comment (drdrsh) [inline src/messages.rs:68]: Is this referenced anywhere?

File: tests/docker/Dockerfile [hunk 2]
```diff
@@ -6,6 +6,8 @@ RUN apt-get update && apt-get install llvm-11 psmisc postgresql-contrib postgres
 RUN cargo install cargo-binutils rustfilt
 RUN rustup component add llvm-tools-preview
 RUN sudo gem install bundler
+RUN wget -O /usr/bin/glauth https://github.com/glauth/glauth/releases/download/v2.3.2/glauth-linux-amd64
+RUN chmod +x /usr/bin/glauth
```
Comment (drdrsh) [inline tests/docker/Dockerfile:10]: These are left overs from the LDAP PR, I assume?

Comment (AndrewJackson2020) [reply tests/docker/Dockerfile]: Correct this is removed now.

File: src/messages.rs (hunk 1 above)
Comment (AndrewJackson2020) [reply src/messages.rs]: Apologies, this was another LDAP thing that I forgot to remove. This is  removed now,

**Comment by AndrewJackson2020** [comment]
@drdrsh Last commit fixed the above 2 issues and cleaned up the tests a bit (implemented proper setup/teardown so a test failure doesnt leave a lingering pgcat server in the test environment, moved the trust pgcat config to a temp file generated by python so that the config is collocated with the relevant test.

**Pull request merged**